### PR TITLE
feat: Custom Generic Spinner Created

### LINKED ISSUE #139: FeedAHelp: Custom Generic Spinner
The Custom Generic Spinner is a user interface (UI) component designed to enhance the user experience of the FeedAHelp web application. The spinner will be used to indicate loading, processing, or waiting states during various interactions within the application.

**Functional Requirements:**

1. Display and Animation:
The spinner should display a visually appealing and engaging animation that indicates activity or loading status to the user.

2. Customizability:
The spinner should be customisable in terms of size, color, and animation style to align with the overall design and branding of the FeedAHelp application.

3. Responsive Design:
The spinner should adapt to different screen sizes and orientations, ensuring a seamless experience across various devices and resolutions.

4. Accessibility:
The spinner should meet accessibility standards, including proper labeling, keyboard navigation, and compatibility with screen readers for users with disabilities.

5. Dynamic Loading States:
The spinner should dynamically appear and disappear based on the application's loading or processing states, triggered by asynchronous actions such as data retrieval or submission.

### PR DESCRIPTION
**Pull request by MNSM92**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #139 

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: apps/feedahelp/src/components/ui/features/hero/Hero.tsx [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+  const [fakeData, setFakeData] = useState("");
+
+  const fetchData = () => {
+    setIsLoading(true);
+    setTimeout(() => {
+      setFakeData("This is Fake Data!!!");
+      setIsLoading(false);
+    }, 5000);
+  };
+
```
Comment (TanvirAlam) [inline apps/feedahelp/src/components/ui/features/hero/Hero.tsx:15]: add this more readable
```
const renderContent = () => {
    if (isLoading) {
      return <CustomSpinner isLoading={isLoading} />;
    }
    return <p>{fakeData}</p>;
  };
  ```

File: apps/feedahelp/src/components/ui/features/hero/Hero.tsx [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
   return (
-    <div className="h-72 mt-28 w-full border border-red-600">
+    <div className="mt-28 h-72 w-full border border-red-600">
       This is the Hero section.
+      <div className="w-20 border-2 bg-green-100">
+        <button className="h-auto w-20" onClick={fetchData}>
+          <p>Click me</p>
+        </button>
+      </div>
+      <CustomSpinner isLoading={isLoading}>
```
Comment (TanvirAlam) [inline apps/feedahelp/src/components/ui/features/hero/Hero.tsx]: replace this
`{renderContent()}`

**Review by TanvirAlam — CHANGES_REQUESTED** [review]
LGTM

**Review by TanvirAlam — APPROVED** [review]
Damn boy nice work

**Review by urmirahman — APPROVED** [review]
LGTM

**Pull request merged**